apps/game_backend/lib/game_backend/units/characters.ex

```diff
@@ -214,4 +216,72 @@ defmodule GameBackend.Units.Characters do
     |> Skin.changeset(attrs)
     |> Repo.insert()
   end
+
+  @doc """
+  Gets a Skin.
+  ## Examples
+      iex> get_skin_by_name(id)
+      {:ok, %Skin{}}
+      iex> get_skin_by_name(wrong_id)
+      {:error, :not_found}
+  """
+  def get_skin_by_name(skin_name) do
+    Repo.get_by(Skin, name: skin_name)
+    |> case do
+      nil -> {:error, :not_found}
+      skin -> {:ok, skin}
+    end
+  end
+
+  def get_skin_purchase_cost_by_currency(skin, currency_id) do
+    purchase_cost =
+      Map.get(skin, :purchase_costs)
```

Comment:
is `purchase_costs` always defined? if so we can just do `skin.purchase_costs`. If not, I think `Enum.find(nil, ...)` will fail so change it for `Map.get(skin, :purchase_costs, [])`

Reply:
Nice catch. Done!